test/functional/main_test.go

```diff
@@ -28,15 +27,34 @@ import (
 	"github.com/Microsoft/hcsshim/internal/uvm"
 	"github.com/Microsoft/hcsshim/internal/winapi"
 
-	testctrd "github.com/Microsoft/hcsshim/test/internal/containerd"
+	"github.com/Microsoft/hcsshim/test/internal/constants"
 	testflag "github.com/Microsoft/hcsshim/test/internal/flag"
+	"github.com/Microsoft/hcsshim/test/internal/layers"
 	"github.com/Microsoft/hcsshim/test/internal/require"
+	"github.com/Microsoft/hcsshim/test/internal/util"
 	testuvm "github.com/Microsoft/hcsshim/test/internal/uvm"
 )
 
 // owner field for uVMs.
 const hcsOwner = "hcsshim-functional-tests"
 
+var (
+	alpineImagePaths = &layers.LazyImageLayers{
```

Comment:
It seems kind of complicated to have this "LazyImageLayers" class when we could just pull the images once in TestMain and have them as globals to use for all the tests or use the supplied image layers. Was there a specific reason we didn't want to do that?

Reply:
to avoid pulling the images if theyre not needed, since not all the tests need a container image
This way if we are just testing uVM functionality, or WCOW (when we eventually fix those test...), or if a special image is needed, like the 70 layer one for vpmem testing, we dont have to pull them all at the beginning just for them not to be used because of the features specified.

Reply:
Do we know how long the image pulls take? IMO if it's not that long, I don't really see a big push for this change here since this is just a test suite. That said, I'm okay with this class if you prefer to keep it.

Reply:
about 5~15 seconds for alpine and nanoserver, and a couple minutes for servercore (if we ever do have to use that in tests)